app/scripts/lib/ppom/ppom-middleware.ts

```diff
@@ -71,65 +89,91 @@ export function createPPOMMiddleware(
       ) {
         // eslint-disable-next-line require-atomic-updates
         const securityAlertId = uuid();
+        let securityAlertResponse: SecurityAlertResponse = {
+          reason: BlockaidResultType.Loading,
+          result_type: BlockaidReason.inProgress,
+          securityAlertId,
+        };
 
         ppomController
           .usePPOM(async (ppom: PPOM) => {
             try {
               const normalizedRequest = normalizePPOMRequest(req);
 
-              const securityAlertResponse = await ppom.validateJsonRpc(
+              securityAlertResponse = await ppom.validateJsonRpc(
                 normalizedRequest,
               );
-
               securityAlertResponse.securityAlertId = securityAlertId;
-              return securityAlertResponse;
-            } catch (error: any) {
+            } catch (error: unknown) {
               sentry?.captureException(error);
-              const errorObject = error as unknown as Error;
-              console.error('Error validating JSON RPC using PPOM: ', error);
-              const securityAlertResponse = {
+              console.error(
+                'Error validating JSON RPC using PPOM: ',
+                typeof error === 'object' || typeof error === 'string'
+                  ? error
+                  : JSON.stringify(error),
```

Comment:
Does not need to be solved in this PR, but why are we doing this serialization?
I get that JSON.stringify(error) when error is an instanceof Error will result in an empty object (as Error properties are not enumerable). But why typeof 'object'?
Wouldn't this be a better approach?
```js
error instanceof Error || typeof error === 'string'
```
For example the above option would allow us to properly log objects with nested arrays of objects.

Also this pattern is repeated a couple of times in this file. Should we at least have a method to `consoleLogSerializedError`? It would just have two args: the error (unknown) and the message to be logged.

Anyways, serliazing an unknown variable properly can be complex, as I am sure I am missing a couple of edge cases.

Reply:
`console.error` outputs interactively expandable representations of input objects. This applies to nested objects/arrays as well, so to my understanding, serialization isn't necessary here unless we're dealing with very large, deeply nested objects. 

`typeof error === 'object'` is because I assumed that we'd want the `console.error` representation for thrown object literals in general, not just `Error` types.

Although I'm not familiar with the logging and monitoring pipeline in the extension, so my assumptions might be off here.

I think a `consoleLogSerializedError` helper method makes a lot of sense. We have [`serializeError`](https://github.com/MetaMask/rpc-errors/blob/aa0754d3574ea3394126b752240caf93a8a9ea10/src/utils.ts#L104) and related methods in `@metamask/rpc-errors` for json-rpc errors. Something similar for browser console and sentry errors would definitely be useful.

Reply:
Oh you are 100% right.. I was thinking about node console and not the browser (which is the env where the extension is executed). My bad 🤦‍♂️ 
@digiwand feel free to ignore my comment about `typeof x === 'object'` vs `instanceof error`.

Yes we should have a common way to not just serialize errors, but also to stack them up (where the stack trace and message of one error does not get lost in the hierarchical error handling). Anyways this out of scope of this PR.